query_generators/anthropic.py

```diff
@@ -95,18 +96,30 @@ def generate_query(
         self.query = ""
         self.reason = ""
 
+        if self.use_public_data:
+            from defog_data.metadata import dbs
+        else:
+            # raise Exception("Replace this with your private data import")
+            from defog_data_private.metadata import dbs
+
         with open(self.prompt_file) as file:
             model_prompt = file.read()
         question_instructions = question + " " + instructions
         if table_metadata_string == "":
-            pruned_metadata_ddl, join_str = prune_metadata_str(
-                question_instructions,
-                self.db_name,
-                self.use_public_data,
-                columns_to_keep,
-                shuffle,
-            )
-            pruned_metadata_str = pruned_metadata_ddl + join_str
+            if columns_to_keep > 0:
+                pruned_metadata_ddl, join_str = prune_metadata_str(
+                    question_instructions,
+                    self.db_name,
+                    self.use_public_data,
+                    columns_to_keep,
+                    shuffle,
+                )
+                pruned_metadata_str = pruned_metadata_ddl + join_str
+            elif columns_to_keep == 0:
```

Comment:
thanks for adding this! sorry I missed this while updating the `columns_to_keep` earlier.

Reply:
All good!